cloudevents/builder.go

```diff
@@ -0,0 +1,65 @@
+package cloudevents
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// CloudEventEncoding is used to tell the builder which encoding to select.
+// the default is Binary.
+type CloudEventEncoding int
+
+const (
+	// Binary
+	BinaryEncoding CloudEventEncoding = 0
+	// Structured
+	StructuredEncoding CloudEventEncoding = 1
+)
+
+// Builder holds settings that do not change over CloudEvents. It is intended
+// to represent a builder of only a single CloudEvent type.
+type Builder struct {
+	Source    string
+	EventType string
+	Encoding  CloudEventEncoding
+}
+
+// Build produces a http request with the constant data embedded in the builder
+// merged with the new data provided in the build function. The request will
+// send a pre-assembled cloud event to the given target. The target is assumed
+// to be a URL with a scheme, ie: "http://localhost:8080"
+func (b *Builder) Build(target string, data interface{}) (*http.Request, error) {
```

Comment:
Is this interface meant to be exposed externally?

If so, can you make this take 0-1 additional arguments (using varargs) to supply a CloudEventsContext. For example, you might want to be able to set the `time` or `id` (v0.2 names) fields on the CloudEvent from upstream data (like data from the PubSub envelope).

If this is meant to be internal and used with the Client, then it might make more sense to de-capitalize this and have the Builder fields be part of the Client.

Reply:
This work was in result of a thought process solving the first pain point, and un-covering a new one.. fixing that. I had thought the builder would solve my problem with the cloudevents lib but it did not, I saw new ways to make it cleaner.

Now that I see the client, I might want to go back and rewrite it with this new use in mind.

Reply:
I would like to leave Build as public just incase someone wants to build contexts and requests without my client.